tests/rptest/tests/retention_policy_test.py

```diff
@@ -443,3 +444,95 @@ def cloud_log_size() -> int:
+        have all SI settings disabled.
+        """
+        total_bytes = 20 * self.segment_size
+        retention_bytes = 10 * self.segment_size
+        cs_housekeeping_interval = 100
+
+        topic = TopicSpec(name=self.topic_name,
+                          partition_count=1,
+                          replication_factor=1,
+                          cleanup_policy=TopicSpec.CLEANUP_DELETE)
+
+        # Cluster-wide settings have remote_read & remote_write enabled
+        self.redpanda.set_cluster_config(
+            {
+                "cloud_storage_enable_remote_write": True,
+                "cloud_storage_enable_remote_read": True,
+                "cloud_storage_housekeeping_interval_ms":
+                cs_housekeeping_interval
+            },
+            expect_restart=True)
+
+        # However, topic settings have main SI functionality disabled
+        self.rpk.create_topic(topic=topic.name,
+                              partitions=topic.partition_count,
+                              replicas=topic.replication_factor,
+                              config={
+                                  "cleanup.policy": topic.cleanup_policy,
+                                  "redpanda.remote.write": "false",
+                                  "redpanda.remote.read": "false",
+                              })
+
+        before_alter = self.rpk.describe_topic_configs(topic.name)
+        assert before_alter['redpanda.remote.write'][0] == 'false'
+        assert before_alter['redpanda.remote.read'][0] == 'false'
+
+        # Write some data to the topic, and assert that nothing has been
+        # written out to S3
+        produce_total_bytes(self.redpanda,
+                            topic=self.topic_name,
+                            partition_index=0,
+                            bytes_to_produce=total_bytes)
+
+        def ntp_in_manifest() -> int:
+            s3_snapshot = S3Snapshot([topic], self.redpanda.s3_client,
+                                     self.s3_bucket_name, self.logger)
+            return s3_snapshot.is_ntp_in_manifest(topic.name, 0)
+
+        # Sleep for 4 cloud_storage_housekeeping_interval_ms, then assert ntp does
+        # not exist in manifest
+        sleep(4 * cs_housekeeping_interval * 0.001)
+        assert not ntp_in_manifest(
+        ), "Segment uploaded, when it should not exist"
+
+        # Now modify topic properties to enable SI read/write for the topic
+        self.rpk.alter_topic_config(topic.name, "redpanda.remote.write",
+                                    "true")
+        self.rpk.alter_topic_config(topic.name, "redpanda.remote.read", "true")
+        after_alter = self.rpk.describe_topic_configs(topic.name)
+        assert after_alter['redpanda.remote.write'][0] == 'true'
+        assert after_alter['redpanda.remote.read'][0] == 'true'
+
+        # Wait for upload to occur first
+        wait_until(lambda: ntp_in_manifest(), timeout_sec=10)
```

Comment:
This check is also implicitly done in the next `wait_until`. I think we can do away with it.

Reply:
No because the next `wait_until` will eventually call:
```
    def manifest_for_ntp(self,
                         topic: str,
                         partition: int,
                         ns: str = 'kafka') -> dict:
        ntp = NTP(ns, topic, partition)
        assert ntp in self.partition_manifests, f'NTP {ntp} not in manifests in S3: ' \
                                                f'{pprint.pformat(self.partition_manifests)}'
        ....
```
which asserts if the NTP is not in the manifest